datafusion/physical-plan/src/common.rs

```diff
@@ -139,17 +140,24 @@ pub fn compute_record_batch_statistics(
 ) -> Statistics {
     let nb_rows = batches.iter().flatten().map(RecordBatch::num_rows).sum();
 
-    let total_byte_size = batches
-        .iter()
-        .flatten()
-        .map(|b| b.get_array_memory_size())
-        .sum();
-
     let projection = match projection {
         Some(p) => p,
         None => (0..schema.fields().len()).collect(),
     };
 
+    let total_byte_size = batches
+        .iter()
+        .flatten()
+        .map(|b| {
+            b.columns()
+                .iter()
+                .enumerate()
+                .filter(|(index, _)| projection.contains(index))
```

Comment:
Might do something like `b.colums()[index]` instead

Reply:
thanks for your advice